### PR TITLE
Fixed webpack setup to seperate compilated client and server code

### LINKED ISSUE #94: Security issue with deploy script
All server side files are exposed when running the deploy script. You can see the file structure afterwards in the following image. The compiled app files are located next to the compiled server files. 

screenshot 2017-06-30 15 34 31

This directory is made public via express.static. So if you use the deploy script to publish your app in production everyone can look up all server side code. As an example you can run npm run deploy and go to localhost:3000/index.js (see below image). This is the compiled server entry.

screenshot 2017-06-30 15 37 37

This is not possible on the deployed heroku version, because there the server side code is compiled to directory **lib**, so it is not located inside the static directory.

### PR DESCRIPTION
**Pull request by jkettmann**
Changed build directories for client and server.

server -> build
client -> build/app

build/app is exposed via express.static, thus only client side code can be accessed publicly

Closes #94

### CODE REVIEW

**Comment by lvarayut** [comment]
Thanks @jkettmann for the awesome PR. Could you change the branch to `canary` instead of `master`, you should be able to do so by clicking the `lvarayut:master` at the top and change it to `lvarayut:canary`?

**Comment by jkettmann** [comment]
@lvarayut Yes, of course. Can you check your heroku deploy? Not sure if I broke that.

File: package.json [hunk 1]
```diff
@@ -97,7 +97,7 @@
     "react-router-relay": "^0.14.0",
     "style-loader": "^0.18.2",
     "url-loader": "^0.5.9",
-    "webpack": "^3.0.0",
+    "webpack": "2",
```
Comment (lvarayut) [inline package.json]: Could you explain why do you downgrade the version of webpack?

File: package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "build-server": "cross-env NODE_ENV=production ./node_modules/.bin/babel ./server --out-dir ./build",
     "lint": "eslint client server",
     "flow": "flow",
-    "heroku-postbuild": "cross-env NODE_ENV=production webpack --config webpack.config.js && cross-env NODE_ENV=production ./node_modules/.bin/babel ./server --out-dir ./lib",
+    "heroku-postbuild": "cross-env NODE_ENV=production webpack --config webpack.config.js && cross-env NODE_ENV=production ./node_modules/.bin/babel ./server --out-dir ./build",
```
Comment (lvarayut) [inline package.json]: Do we need to change the output path of the Heroku app?

File: webpack.config.js [hunk 3]
```diff
@@ -41,6 +43,7 @@ if (process.env.NODE_ENV === 'production') {
     'webpack-dev-server/client?http://localhost:3000',
     'webpack/hot/only-dev-server'
   ];
+  outputPath = path.join(__dirname, 'app');
```
Comment (lvarayut) [inline webpack.config.js]: Do we need to use the different path for the development environment?

**Review by lvarayut — COMMENTED** [review]
👍 , just left some curious comments 😃

File: package.json (hunk 1 above)
Comment (jkettmann) [reply package.json]: Actually I couldn't start with webpack 3. Doesn't have anything to do with this issue though.

File: package.json (hunk 2 above)
Comment (jkettmann) [reply package.json]: You're right. This is not necessary.

File: webpack.config.js (hunk 3 above)
Comment (jkettmann) [reply webpack.config.js]: You're right again. Is the contentBase and static server in server/index.js necessary (if I remove them everything seems to be still working). I guess that's why I used a different path for development.

```
const relayServer = new WebpackDevServer(webpack(webpackConfig), {
    contentBase: '/build/',

...

relayServer.use('/', express.static(path.join(__dirname, '../build/app')));
```

File: package.json (hunk 1 above)
Comment (ncrmro) [reply package.json]: @jkettmann Also having trouble starting new webpack 3 something along these lines?

```js
ERROR in   TypeError: Cannot read property 'request' of undefined
  
  - ExternalModuleFactoryPlugin.js:37 handleExternals
    [relay-fullstack]/[html-webpack-plugin]/[webpack]/lib/ExternalModuleFactoryP    lugin.js:37:33
  
....
```

Comment (jkettmann) [reply package.json]: @ncrmro [Have a look at this comment](https://github.com/jantimon/html-webpack-plugin/pull/707#issuecomment-310892004)

**Comment by jkettmann** [comment]
@lvarayut I removed the unnecessesary changes.

File: package.json (hunk 1 above)
Comment (lvarayut) [reply package.json]: @ncrmro and @jkettmann, those are very useful information

**Comment by lvarayut** [comment]
Thanks @jkettmann 👍

**Pull request merged**